File: demo/test/ViewController.m
//
//  ViewController.m
//  test
//
//  Created by 全尼古拉斯 on 2020/9/9.
//  Copyright © 2020 全尼古拉斯. All rights reserved.
//

#import "ViewController.h"
#import <XBInterActiveAdSDK/XBInterActiveAdSDK.h>
@interface ViewController ()

@end

@implementation ViewController

//let appKey = "demo"
//let appSecret = "ac4a10da9e7f62adb59dbe7f62adb59dbe770e8d"
//let soltId: Int64 = 1
- (void)viewDidLoad {
    [super viewDidLoad];
    UIButton *btn = [UIButton buttonWithType: UIButtonTypeRoundedRect];
    btn.frame = CGRectMake(50, 50, 300, 60);
    [btn setBackgroundColor:UIColor.redColor];
    [btn setTitle:@"clear Ad" forState:UIControlStateNormal];
    [btn addTarget:self action:@selector(onClickClear:) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:btn];

    [[XBInterActiveAdManager shared] startWithAppKey:@"vnntest0529" appSecret:@"9a19fab1935aba50f1fd5a6bdb442172" complete:^(XBError *error) {
        dispatch_async(dispatch_get_main_queue(), ^{
            [self show];
        });
        
    }];
}

- (void)onClickClear: (UIButton *)btn
{
    [[XBInterActiveAdManager shared] clearAdWithTag:1024 customView:self.view];
}

- (void)show
{
    [[XBInterActiveAdManager shared] showAdWithWidgetId:@"986" scale:1 origin:CGPointMake(200, 500) tag: 1024 customView:self.view onShowAction:^{
        NSLog(@"Ad show");
    } onClickAction:^{
        NSLog(@"ad clicked");
    } onCloseAction:^{
        NSLog(@"ad close");
    } onLoadFailed:^(XBError *error) {
        NSLog(@"ad failed");
        NSLog(error.description);
    } onClearAction:^{
        NSLog(@"ad clear");
    }];
}

@end

File: swift demo/test2/test2/ViewController.swift
//
//  ViewController.swift
//  test2
//
//  Created by 全尼古拉斯 on 2020/9/9.
//  Copyright © 2020 全尼古拉斯. All rights reserved.
//

import UIKit
import XBInterActiveAdSDK

let appKey = "vnntest0529"
let appSecret = "9a19fab1935aba50f1fd5a6bdb442172"
let widgetId = "986"


let soltId: Int64 = 1

class ViewController: UIViewController {
    var customView: UIView?
    var button: UIButton?
    override func viewDidLoad() {
        super.viewDidLoad()
        
        button = UIButton.init(frame: CGRect.init(x: 50, y: 100, width: 300, height: 60))
        button?.setTitle("close Ad Action", for: .normal)
        button?.addTarget(self, action: #selector(onClickClose), for: .touchUpInside)
        button?.backgroundColor = UIColor.red
        self.view.addSubview(button!)
        
        
        customView = UIView.init(frame: CGRect.init(x: 0, y: UIScreen.main.bounds.height / 2, width: UIScreen.main.bounds.width, height: UIScreen.main.bounds.height / 2))
        self.view.addSubview(customView!)
        
        XBInterActiveAdManager.shared.start(appKey: appKey, appSecret: appSecret, complete: { (error) in
            if error == nil {
                print("注册成功")
                DispatchQueue.main.async {
                    self.show()
                    print("")
                }
            }
        })
        
    }

    @objc func onClickClose() {
        XBInterActiveAdManager.shared.clearAd(tag: 1024, customView: customView!)
    }
    
    func show() {
        
        XBInterActiveAdManager.shared.showAd(widgetId: widgetId, scale: 2, tag: 1024, customView: customView!, onShowAction: {
            print("广告展示")
        }, onClickAction: {
            print("用户点击")
        }, onCloseAction: {
            print("用户关闭")
        }, onLoadFailed: { (error) in
            print("显示失败")
        }, onClearAction: {
            print("主动清除广告")
        })
    }
}
